bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/DiscoveryListener.java

```diff
@@ -54,52 +60,68 @@ public DiscoveryListener() throws IOException, SocketException {
+            localDatagramSocket.setBroadcast(true);
+            dSocket = localDatagramSocket;
             rcvBuffer = new byte[256];
             rcvPacket = new DatagramPacket(rcvBuffer, rcvBuffer.length);
             bcastAddress = InetAddress.getByName(BCAST_ADDRESS);
             bcastBuffer = POLL_MESSAGE.getBytes(StandardCharsets.US_ASCII);
             bcastPacket = new DatagramPacket(bcastBuffer, bcastBuffer.length, bcastAddress, BAF_PORT);
-        } catch (UnknownHostException uhe) {
-            logger.warn("UnknownHostException sending poll request for fans: {}", uhe.getMessage(), uhe);
+        } catch (UnknownHostException | SocketException | SecurityException e) {
+            logger.warn("Unexpected exception sending poll request for fans: {}", e.getMessage(), e);
         }
     }
 
     public BigAssFanDevice waitForMessage() throws IOException, SocketTimeoutException {
         // Wait to receive a packet
-        rcvPacket.setLength(rcvBuffer.length);
-        dSocket.receive(rcvPacket);
+        DatagramPacket localPacket = rcvPacket;
+        DatagramSocket localDatagramSocket = dSocket;
+
+        if (localPacket != null) {
+            localPacket.setLength(rcvBuffer.length);
+        }
+
+        if (localDatagramSocket != null && localPacket != null) {
+            localDatagramSocket.receive(rcvPacket);
 
-        // Process the received packet
-        device.reset();
-        device.setIpAddress(rcvPacket.getAddress().getHostAddress());
-        String message = (new String(rcvBuffer, 0, rcvPacket.getLength()));
-        device.setDiscoveryMessage(message);
-        logger.debug("RECEIVED packet of length {} from {}: {}", message.length(), device.getIpAddress(), message);
+            // Process the received packet
+            device.reset();
+
+            String address = localPacket.getAddress().getHostAddress();
+            if (address != null) {
+                device.setIpAddress(address);
+            }
+
+            String message = (new String(rcvBuffer, 0, localPacket.getLength()));
+            device.setDiscoveryMessage(message);
+            logger.debug("RECEIVED packet of length {} from {}: {}", message.length(), device.getIpAddress(), message);
+        }
 
         return device;
     }
 
     public void pollForDevices() {
-        if (dSocket == null) {
+        DatagramSocket localDatagramSocket = dSocket;
+        if (localDatagramSocket == null) {
             logger.debug("Socket is null in discoveryListener.pollForDevices()");
             return;
         }
 
         logger.debug("Sending poll request for fans: {}", POLL_MESSAGE);
         try {
-            dSocket.send(bcastPacket);
-        } catch (IOException ioe) {
-            logger.warn("IOException sending poll request for fans: {}", ioe.getMessage(), ioe);
+            localDatagramSocket.send(bcastPacket);
+        } catch (IllegalArgumentException | SecurityException | IOException ioe) {
```

Comment:
Since no longer only catching `IOException`, probably name could be more generic:
```suggestion
        } catch (IllegalArgumentException | SecurityException | IOException e) {
```